src/tree/fleetTreeItem.ts

```diff
@@ -0,0 +1,98 @@
+import { AzExtParentTreeItem, AzExtTreeItem, IActionContext } from "@microsoft/vscode-azext-utils";
+import { CloudExplorerV1 } from "vscode-kubernetes-tools-api";
+import { assetUri } from "../assets";
+import { DefinedResourceWithGroup, getFleetMembers } from "../commands/utils/azureResources";
+import { SubscriptionTreeNode } from "./subscriptionTreeItem";
+import { createClusterTreeNode } from "./aksClusterTreeItem";
+import { ReadyAzureSessionProvider } from "../auth/types";
+import { failed } from "../commands/utils/errorable";
+import { window } from "vscode";
+import { parseResource } from "../azure-api-utils";
+
+// The de facto API of tree nodes that represent individual AKS clusters.
+// Tree items should implement this interface to maintain backward compatibility with previous versions of the extension.
+export interface FleetTreeNode {
+    readonly nodeType: "fleet";
+    readonly resourceType: string;
+    readonly armId: string;
+    readonly name: string;
+    readonly subscriptionTreeNode: SubscriptionTreeNode;
+    readonly subscriptionId: string;
+    readonly resourceGroupName: string;
+}
+
+export function createFleetTreeNode(
+    parent: AzExtParentTreeItem & SubscriptionTreeNode,
+    sessionProvider: ReadyAzureSessionProvider,
+    subscriptionId: string,
+    fleetResource: DefinedResourceWithGroup,
+): FleetTreeItem {
+    return new FleetTreeItem(parent, sessionProvider, subscriptionId, fleetResource);
+}
+
+class FleetTreeItem extends AzExtParentTreeItem implements FleetTreeNode {
+    private readonly sessionProvider: ReadyAzureSessionProvider;
+    public readonly subscriptionTreeNode: SubscriptionTreeNode;
+    public readonly armId: string;
+    public readonly resourceGroupName: string;
+    public readonly name: string;
+    private readonly fleetResource: DefinedResourceWithGroup;
+
+    constructor(
+        parent: AzExtParentTreeItem & SubscriptionTreeNode,
+        sessionProvider: ReadyAzureSessionProvider,
+        readonly subscriptionId: string,
+        readonly fleet: DefinedResourceWithGroup,
```

Comment:
should not be marked readonly. only internal.
probably, read about how class constructor work in typespec :)